projects/packages/my-jetpack/src/products/class-product.php

```diff
@@ -379,12 +392,42 @@ public static function get_standalone_info() {
 	}
 
 	/**
-	 * Checks whether the site has a paid plan for the product
-	 * This ignores free products, it only checks if there is a purchase that supports the product
+	 * Checks whether the site has a paid plan for the product.
+	 *
+	 * This function relies on the product's `$feature_identifying_paid_plan` and `get_paid_plan_product_slugs()` function.
+	 * If the product does not define a `$feature_identifying_paid_plan`, be sure the product includes functions for both
+	 * `get_paid_plan_product_slugs()` and `get_paid_bundles_that_include_product()` which return all the product slugs and
+	 * bundle slugs that include the product, respectively.
 	 *
 	 * @return boolean
 	 */
 	public static function has_paid_plan_for_product() {
+		// First check site features (if there's a feature that identifies the paid plan)
+		if ( static::$feature_identifying_paid_plan ) {
```

Comment:
Seems like having this check fulfills most of what [this issue](https://github.com/Automattic/jetpack-marketing/issues/905) needs to be fixed 😄  Once this is done, it should be a quick patch to get this one done finally 🥳

Reply:
@grzegorz-cp would you be able to follow up on this, and get [the issue](https://github.com/Automattic/jetpack-marketing/issues/905) done?